metadata-ingestion/src/datahub/ingestion/source/cassandra.py

```diff
@@ -0,0 +1,592 @@
+    "tables": "tables",
+    "views": "views",
+    "columns": "columns",
+}
+# these column names are present on the system_schema tables
+CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES = {
+    "keyspace_name": "keyspace_name",  # present on all tables
+    "table_name": "table_name",  # present on tables table
+    "column_name": "column_name",  # present on columns table
+    "column_type": "type",  # present on columns table
+    "view_name": "view_name",  # present on views table
+    "base_table_name": "base_table_name",  # present on views table
+    "where_clause": "where_clause",  # present on views table
+}
+
+
+# -------------------------------------------------- queries -------------------------------------------------- #
+# get all keyspaces
+GET_KEYSPACES_QUERY = f"SELECT * FROM {SYSTEM_SCHEMA_KESPACE_NAME}.{CASSANDRA_SYSTEM_SCHEMA_TABLES['keyspaces']}"
+# get all tables for a keyspace
+GET_TABLES_QUERY = f"SELECT * FROM {SYSTEM_SCHEMA_KESPACE_NAME}.{CASSANDRA_SYSTEM_SCHEMA_TABLES['tables']} WHERE {CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES['keyspace_name']} = %s"
+# get all columns for a table
+GET_COLUMNS_QUERY = f"SELECT * FROM {SYSTEM_SCHEMA_KESPACE_NAME}.{CASSANDRA_SYSTEM_SCHEMA_TABLES['columns']} WHERE {CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES['keyspace_name']} = %s AND {CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES['table_name']} = %s"
+# get all views for a keyspace
+GET_VIEWS_QUERY = f"SELECT * FROM {SYSTEM_SCHEMA_KESPACE_NAME}.{CASSANDRA_SYSTEM_SCHEMA_TABLES['views']} WHERE {CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES['keyspace_name']} = %s"
+
+
+# -------------------------------------------------- source config and reporter -------------------------------------------------- #
+
+# config
+class CassandraSourceConfig(PlatformInstanceConfigMixin, EnvConfigMixin):
+    contact_point: str = Field(
+        default="localhost",
+        description="The cassandra instance contact point domain (without the port).",
+    )
+    port: str = Field(default="10350", description="The cassandra instance port.")
+    username: str = Field(
+        default="",
+        description=f"The username credential. Ensure user has read access to {SYSTEM_SCHEMA_KESPACE_NAME}.",
+    )
+    password: str = Field(default="", description="The password credential.")
+    keyspace_pattern: AllowDenyPattern = Field(
+        default=AllowDenyPattern.allow_all(),
+        description="regex patterns for tables to filter in ingestion.",
+    )
+
+
+# source reporter
+@dataclass
+class CassandraSourceReport(SourceReport):
+    filtered: List[str] = field(default_factory=list)
+
+    def report_dropped(self, index: str) -> None:
+        self.filtered.append(index)
+
+
+# -------------------------------------------------- main source class -------------------------------------------------- #
+@platform_name("Cassandra")
+@config_class(CassandraSourceConfig)
+@support_status(SupportStatus.TESTING)
+@capability(SourceCapability.PLATFORM_INSTANCE, "Enabled by default")
```

Comment:
I'm a bit confused by what several of the capabilities mean.
I kept it simple but I could get suggestions for extra capabilities I should document and why those make sense that'd be helpful.

Reply:
So this is purely documentation?